### ISSUE TITLE
How can I Stop apartment from interfering with my Redshift connections (setting search path)?

### ISSUE THREAD

**Issue by NielsKSchjoedt** [influitive/apartment#394]
I have a rails App which uses apartment with a local postgres database. I would now like to setup a new model, which connects to redshift instead. For that I use the gem https://github.com/aamine/activerecord4-redshift-adapter. I set it up like this:

```yaml
# secrets.yml
redshift_config:
    host: hostname.xxxxxxxxx.eu-central-1.redshift.amazonaws.com
    port: 5439
    user: user_name
    password: pass
    database: db_name
    adapter: redshift
    encoding: utf8
    pool: 3
    timeout: 5000
```
My model looks like:

```ruby
module Tracking
  class Event < ActiveRecord::Base
    establish_connection Rails.application.secrets.redshift_config
    self.table_name = :events
  end
end
```

Now then doing: `Tracking::Event.connection.execute("show search_path").first` I get `{"search_path"=>"da, extensions"}` (which is my default search path for Apartment). But in fact, the user I connect as, has a search_path of: `$user, public, atomic` which I would of cause like to ALWAYS keep when using that exact model/connection. Any suggestions on how I could achieve this?

**Comment by mikecmpbll** [comment]
is Tracking an excluded model?

**Comment by NielsKSchjoedt** [comment]
Yes from a usage perspective it should not be using tenants - But "just" putting it into excluded_models will not work right? From what I see in the documentation excluded models will still set search_path to the using the `default_schema` and `persistent_schemas` configs, which again has nothing to do with that particular table/database in redshift.

**Comment by mikecmpbll** [comment]
to be honest, i'm not precisely sure why what you're doing at the moment isn't working. when apartment switches tenants (pg), it modifies the schema_search_path on the connection (ActiveRecord::Base.connection by default). you're establishing a separate connection on that model, so apartment should have nothing to do with it.

have you confirmed that apartment is affecting this search_path and that it's not something else?

**Comment by NielsKSchjoedt** [comment]
Doh. I just realised I made a mistake. I was testing the connection to redshift locally (in development). Then afterwood I deployed to production and tried to query real data. Then I threw the missing schema error indicating wrong schema path. I just realised now, that the connection for the model was by mistake done to my primary postgresql database, since my secrets.yml is overwritten on deployment, and the connection settings missing from that one, so there were some kind of fallback to the database.yml. My mistake. It works fine out of the box. Sorry again. :-)
